rules/framework.bzl

```diff
@@ -40,27 +40,48 @@ _APPLE_FRAMEWORK_PACKAGING_KWARGS = [
     "exported_symbols_lists",
 ]
 
-def apple_framework(name, apple_library = apple_library, **kwargs):
+def apple_framework(
+        name,
+        apple_library = apple_library,
+        infoplists = [],
+        infoplists_by_build_setting = {},
+        xcconfig = {},
+        xcconfig_by_build_setting = {},
+        **kwargs):
     """Builds and packages an Apple framework.
 
     Args:
         name: The name of the framework.
         apple_library: The macro used to package sources into a library.
+        infoplists: A list of Info.plist files to be merged into the app clip.
+        infoplists_by_build_setting: A dictionary of infoplists grouped by bazel build setting.
+
+                                     Each value is applied if the respective bazel build setting
+                                     is resolved during the analysis phase.
+
+                                     If '//conditions:default' is not set the value in 'infoplists'
+                                     is set as default.
+        xcconfig: A dictionary of xcconfigs to be applied to the app clip by default.
+        xcconfig_by_build_setting: A dictionary of xcconfigs grouped by bazel build setting.
+
+                                   Each value is applied if the respective bazel build setting
+                                   is resolved during the analysis phase.
+
+                                   If '//conditions:default' is not set the value in 'xcconfig'
+                                   is set as default.
         **kwargs: Arguments passed to the apple_library and apple_framework_packaging rules as appropriate.
     """
     framework_packaging_kwargs = {arg: kwargs.pop(arg) for arg in _APPLE_FRAMEWORK_PACKAGING_KWARGS if arg in kwargs}
     kwargs["enable_framework_vfs"] = kwargs.pop("enable_framework_vfs", True)
 
-    infoplists_by_build_setting = kwargs.pop("infoplists_by_build_setting", {})
-    default_infoplists = kwargs.pop("infoplists", [])
-    infoplists = None
-    if len(infoplists_by_build_setting.values()) > 0 or len(default_infoplists) > 0:
+    merged_infoplists = None
```

Comment:
`merged_infoplists` is not being set anywhere IIUC?

Reply:
```
    apple_framework_packaging(
        name = name,
        framework_name = library.namespace,
        infoplists = merged_infoplists,
```

Gets set on the framework

Reply:
yep but if I followed the code paths correctly it's set to `None` above and then passed as `None` to the fmw right? I don't see it receive any non-`None` value anywhere

Reply:
🤦🏼 ah right, i've updated it to get set as the result of the `process_infoplist`!